ws/src/session.rs

```diff
@@ -182,6 +182,10 @@ impl<M: core::Metadata, S: core::Middleware<M>> Session<M, S> {
 		req.header("origin").map(|x| &x[..])
 	}
 
+	fn read_host<'a>(&self, req: &'a ws::Request) -> Option<&'a [u8]> {
+		req.header("host").map(|x| &x[..])
```

Comment:
doesn't this :point_down: work?

```suggestion
		req.header("host").as_ref()
```